src/storage/SparqlResourceStore.ts

```diff
@@ -0,0 +1,753 @@
+   * @param type - Type of the quads to indicate in the possible error.
+   * @param quads - Incoming quads.
+   *
+   * @throws {@link ConflictHttpError}
+   * If one or more quads are not in the default graph.
+   */
+  private ensureValidQuads(type: string, quads?: Quad[]): Quad[] {
+    if (quads) {
+      if (!quads.every((x): any => inDefaultGraph(x))) {
+        throw new ConflictHttpError(`All ${type} quads should be in the default graph.`);
+      }
+      return quads;
+    }
+    return [];
+  }
+
+  /**
+   * Helper function to get the parent container URI of a container URI.
+   * @param containerURI - Incoming container URI.
+   */
+  private getParentContainer(containerURI: string): string {
+    const [ , parentContainerURI ] = /^(.*\/)[^/]+\/$/u.exec(containerURI) ?? [];
+    if (typeof parentContainerURI !== 'string') {
+      throw new Error('Invalid containerURI passed.');
+    }
+    return parentContainerURI;
+  }
+
+  /**
+   * Helper function without extra validation to send a query to the SPARQL endpoint.
+   * @param sparqlQuery - Query to send.
+   */
+  private async sendSparqlQuery(sparqlQuery: SparqlQuery): Promise<any> {
+    // Generate the string SPARQL query from the SparqlQuery object.
+    const generator = new Generator();
+    const generatedQuery = generator.stringify(sparqlQuery);
+
+    // Send the HTTP request.
+    const init = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/sparql-query',
+        Accept: 'application/json',
+      },
+      body: generatedQuery,
+    };
+    const request = new Request(this.sparqlEndpoint);
+    const response = await fetch(request, init);
+
+    // Check if the server returned an error and return the json representation of the result.
+    if (response.status >= 400) {
+      throw new Error('Bad response from server');
+    }
+    return response.json();
+  }
+
+  /**
+   * Helper function without extra validation to send an update query to the SPARQL endpoint.
+   * @param sparqlQuery - Query to send. In the case of a string, the literal input is forwarded.
+   */
+  private async sendSparqlUpdate(sparqlQuery: SparqlQuery | string): Promise<void> {
+    // Generate the string SPARQL query from the SparqlQuery object if it is passed as such.
+    let generatedQuery;
+    if (typeof sparqlQuery === 'string') {
+      generatedQuery = sparqlQuery;
+    } else {
+      const generator = new Generator();
+      generatedQuery = generator.stringify(sparqlQuery);
+    }
+
+    // Send the HTTP request.
+    const init = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/sparql-update',
+      },
+      body: generatedQuery,
+    };
+    const request = new Request(this.sparqlEndpoint);
+    const response = await fetch(request, init);
+
+    // Check if the server returned an error.
+    if (response.status >= 400) {
+      throw new Error('Bad response from server');
+    }
```

Comment:
Since this is identical to the other call I would put it in 1 funtion to make sure we keep handling errors consistently.